pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -423,21 +426,29 @@ protected void setServletContext(ServletContext servletContext) {
         this.servletContext = servletContext;
     }
 
-    protected CompletableFuture<PartitionedTopicMetadata> getPartitionedTopicMetadataAsync(
-            TopicName topicName, boolean authoritative, boolean checkAllowAutoCreation) {
+    protected PartitionedTopicMetadata getPartitionedTopicMetadata(TopicName topicName,
+                                                                   boolean authoritative,
+                                                                   boolean checkAllowAutoCreation) {
         try {
-            validateClusterOwnership(topicName.getCluster());
-        } catch (Exception e) {
-            return FutureUtil.failedFuture(e);
+            return getPartitionedTopicMetadataAsync(topicName, authoritative, checkAllowAutoCreation)
```

Comment:
Maybe we can use `PulsarWebResource#sync` to reduce duplicate code.

Reply:
fixed